creator-node/src/services/stateMachineManager/index.js

```diff
@@ -49,54 +52,70 @@ class StateMachineManager {
     )
 
-    // Upon completion, make queue jobs record metrics and enqueue other jobs as necessary
-    const queueNameToQueueMap = {
-      [QUEUE_NAMES.MONITOR_STATE]: {
-        queue: monitorStateQueue,
-        maxWaitingJobs: 10
-      },
-      [QUEUE_NAMES.FIND_SYNC_REQUESTS]: {
-        queue: findSyncRequestsQueue,
-        maxWaitingJobs: 10
-      },
-      [QUEUE_NAMES.FIND_REPLICA_SET_UPDATES]: {
-        queue: findReplicaSetUpdatesQueue,
-        maxWaitingJobs: 10
-      },
-      [QUEUE_NAMES.MANUAL_SYNC]: {
-        queue: manualSyncQueue,
-        maxWaitingJobs: 1000
-      },
-      [QUEUE_NAMES.RECURRING_SYNC]: {
-        queue: recurringSyncQueue,
-        maxWaitingJobs: 100000
-      },
-      [QUEUE_NAMES.UPDATE_REPLICA_SET]: {
-        queue: updateReplicaSetQueue,
-        maxWaitingJobs: 10000
-      },
-      [QUEUE_NAMES.RECOVER_ORPHANED_DATA]: {
-        queue: recoverOrphanedDataQueue,
-        maxWaitingJobs: 10
+    if (cluster.worker?.id === 1) {
+      SyncRequestDeDuplicator.clear()
+
+      // Upon completion, make queue jobs record metrics and enqueue other jobs as necessary
+      const queueNameToQueueMap = {
+        [QUEUE_NAMES.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP]: {
+          queue: cNodeEndpointToSpIdMapQueue,
+          maxWaitingJobs: 10
+        },
+        [QUEUE_NAMES.MONITOR_STATE]: {
+          queue: monitorStateQueue,
+          maxWaitingJobs: 10
+        },
+        [QUEUE_NAMES.FIND_SYNC_REQUESTS]: {
+          queue: findSyncRequestsQueue,
+          maxWaitingJobs: 10
+        },
+        [QUEUE_NAMES.FIND_REPLICA_SET_UPDATES]: {
+          queue: findReplicaSetUpdatesQueue,
+          maxWaitingJobs: 10
+        },
+        [QUEUE_NAMES.MANUAL_SYNC]: {
+          queue: manualSyncQueue,
+          maxWaitingJobs: 1000
+        },
+        [QUEUE_NAMES.RECURRING_SYNC]: {
+          queue: recurringSyncQueue,
+          maxWaitingJobs: 100000
+        },
+        [QUEUE_NAMES.UPDATE_REPLICA_SET]: {
+          queue: updateReplicaSetQueue,
+          maxWaitingJobs: 10000
+        },
+        [QUEUE_NAMES.RECOVER_ORPHANED_DATA]: {
+          queue: recoverOrphanedDataQueue,
+          maxWaitingJobs: 10
+        }
+      }
+      for (const queueName of Object.keys(queueNameToQueueMap)) {
+        const queueEvents = new QueueEvents(queueName, {
+          connection: {
+            host: config.get('redisHost'),
+            port: config.get('redisPort')
+          }
+        })
+        queueEvents.on(
+          'completed',
+          makeOnCompleteCallback(
+            queueName,
+            queueNameToQueueMap,
+            prometheusRegistry
+          ).bind(this)
+        )
+
+        // Update the mapping in this StateMachineManager whenever a job successfully fetches it
+        if (queueName === QUEUE_NAMES.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP) {
+          queueEvents.on(
+            'completed',
+            this.updateMapOnMapFetchJobComplete.bind(this)
```

Comment:
I find that syntax more confusing - this is pretty standard for a regular callback to just pass it instead of another anonymous function. either way, you can find the function signature [here](https://api.docs.bullmq.io/interfaces/QueueEventsListener.html#completed). it's different from what you linked because this is for QueueEvents vs Worker events (it's a little confusing that the APIs differ haha). this is tested and working to update the mapping btw

Reply:
ah cool